src/loader/BlendLoader.py

```diff
@@ -73,28 +122,115 @@ def run(self):
         else:
             entities = None
 
-        previously_loaded_objects = set(bpy.context.scene.objects)
+        data_block_name = load_from.strip("/")
 
-        with bpy.data.libraries.load(path) as (data_from, data_to):
+        with bpy.data.libraries.load(path) as (blend_file_data, _):
             # check if defined ID is supported
-            if load_from in self.known_datablock_names.keys():
+            if data_block_name in self.known_datablock_names:
+                attr_name = self._find_datablock_name_match_in_blendfile(
+                    blend_file_data, data_block_name)
+
                 # if some regex was specified, get corresponding matching entity's names
                 if entities is not None:
-                    entities_to_load = [item for item in getattr(data_from, self.known_datablock_names[load_from])
-                                        if re.fullmatch(entities, item) is not None]
+                    entities_to_load = [
+                        item for item in getattr(blend_file_data, attr_name)
+                        if re.fullmatch(entities, item) is not None
+                    ]
+
                 # get all entity's names if not
                 else:
-                    entities_to_load = getattr(data_from, self.known_datablock_names[load_from])
+                    entities_to_load = getattr(blend_file_data, attr_name)
                 # load entities
                 for entity_to_load in entities_to_load:
-                    bpy.ops.wm.append(filepath=os.path.join(path, load_from, entity_to_load),
-                                      filename=entity_to_load,
-                                      directory=os.path.join(path + load_from))
+
+                    # remove the earlier existing resource with same name
+                    if entity_to_load in bpy.data.objects:
```

Comment:
Why would we remove existing objects in the scene, if they happen to have the same name?

Can we not just rename the newly loaded objects, so that they don't clash?

Reply:
They are renamed automatically, no?

Reply:
I was thinking about preventing duplicates, but you are right as its quite possible that different objects can have same name or even multiple same objects are needed. Is renaming old objects fine? new entities are renamed automatically but how do we know that which is the new instance as we don't get the name after loading and not every type is retrievable from context?

Reply:
I think we assume that new entities are renamed by default in all loaders. Maybe we should leave it that way since its Blenders default behaviour?

Reply:
how about camera? how do we find the name of new added camera? of course we can keep a list of old cameras but soon it will clutter up once we have other types that we might want to have a specific behavior for. What do you think of the current committed solution?

Reply:
> we not just rename the newly loaded objects

If I understand you correctly, we just let Blender itself rename them as its always doing.

> how about camera?

Hard to say, but I think we should stick to default naming convention and allow Blender to rename them. Overall it is the same situation as with objects, and it was not that much of a problem. But cameras are unique in many regards when compared to objects, though.

If it is a problem (but it will be only when we have a valid case, which can be an example in your PR, by the way :D), then from the top of my head, we could support a custom function that assigns some sort of value as ID (path to this particular Camera instance, including blend file name and cameras original name). This "crutch" and the situation with multiple cameras in general calls for creating a Camera getter Provider, which would be a must-have, but if @themasterlink agrees, then we must decide if its to be included in this PR.

For now I would suggest leaving the names as they are so we do not introduce a breaking change/behaviour in regards to them, since this convention is supported for all types, at least for now.

Reply:
To put it simply I do not agree. There is nothing gained by introducing several cameras, each camera parameter can be key framed and using several just makes the whole problem more complicated without any real application.

So, as cameras are unique in how they work. I would suggest that a newly loaded camera always replaces our existing camera. Else, no one would have loaded the camera in the first place. As I said here: https://github.com/DLR-RM/BlenderProc/pull/145#discussion_r564292109

And about the deletion, there should be no deletion in a loading module for existing blender objects, that is against the expected behavior of a loading module. If two objects have the same name blender will automatically change the name of the new objects accordingly.

Reply:
I do not think a loader should replace/delete anything at all, it would be unexpected or sometimes even "hidden" behaviour. Instead docu should explain why having multiple cameras is a bad idea and discourage doing so. It would leave room for coding an `keep cameras` param if user needs it. from the side of the config interface and user perspective cameras should not be any different than objects or materials when loading.